lib/tasks/export.rake

```diff
@@ -76,6 +77,34 @@ def export_question(csv, node, level = nil)
   end
 end
 
+def export_email(email, prefix, zos)
+  subject = (email.subject || 'no subject').gsub(' ', '_').gsub(/[^0-9a-z_]/i, '')
+  # some subjects have special cahrs in them, which messes with the wkhtmltopdf bash
+  # truncate at 200 char or filename migh tbe too long
+  filename = [email.sent_at.to_formatted_s(:number), subject].join('_')[0..200]
+  mk_zip_entry(zos, "#{prefix}/email/#{filename}.eml", email.sent_at) do
+    zos << email.raw_source
+  end
+
+  # pdf creation
+  begin
+    temp_eml = Tempfile.new("#{filename}.eml")
+    sanitized_email_data = email.raw_source
+      .gsub(/<img.*?>/m, '') # generally the images link to expired s3 sources which fail the conversion
+      .gsub('=0D', '') # Odd artifacts, maybe from copying stuff from office?
+      .gsub('=3D', '=')
+    temp_eml.write(sanitized_email_data)
+    temp_eml.close
+    temp_pdf = Tempfile.new("#{filename}.pdf")
+    EmlToPdf.convert(temp_eml.path, temp_pdf.path)
+    mk_zip_entry(zos, "#{prefix}/email/#{filename}.pdf", email.sent_at) do
+      zos << temp_pdf.read
```

Comment:
this is going to load the whole PDF file into memory but I guess they should be pretty small.

Reply:
🤷‍♂ its pretty small